src/main/drivers/serial_uart_impl.h

```diff
@@ -17,10 +17,154 @@
 
 #pragma once
 
-// device specific uart implementation is defined here
+// Configuration constants
+
+#if defined(STM32F1)
+#define UARTDEV_COUNT_MAX 3
+#define UARTHARDWARE_MAX_PINS 3
+#ifndef UART_RX_BUFFER_SIZE
+#define UART_RX_BUFFER_SIZE     256
+#endif
+#ifndef UART_TX_BUFFER_SIZE
+#define UART_TX_BUFFER_SIZE     256
+#endif
+#elif defined(STM32F3)
+#define UARTDEV_COUNT_MAX 5
+#define UARTHARDWARE_MAX_PINS 4
+#ifndef UART_RX_BUFFER_SIZE
+#define UART_RX_BUFFER_SIZE     256
+#endif
+#ifndef UART_TX_BUFFER_SIZE
+#define UART_TX_BUFFER_SIZE     256
+#endif
+#elif defined(STM32F4)
+#define UARTDEV_COUNT_MAX 6
+#define UARTHARDWARE_MAX_PINS 4
+#ifndef UART_RX_BUFFER_SIZE
+#define UART_RX_BUFFER_SIZE     512
```

Comment:
I think the comment has been outdated for some time. The buffer indices in `serialPort_t` (`{rx,tx}Buffer{Head,Tail}`) are `uint32_t` (overkill?), and return type for `serialRxBytesWaiting` and `serialTxBytesFree` are also `uint32_t`. Is there anywhere else the 256 limit is imposed?